File: frontend/src/features/admin/types/metricas.ts
export interface MetricasResponse {
  total_alumnos: number;
  alumnos_activos: number;
  porcentaje_riesgo: number;
  promedio_progreso: number;
  total_docentes: number;
  total_materias_activas: number;
  total_carreras_activas: number;
}

export interface AccionesPorDia {
  fecha: string;
  total: number;
}

export interface EstadoComunicacion {
  estado: string;
  cantidad: number;
  docente_nombre?: string;
  materia_nombre?: string;
}

export interface InteraccionDocente {
  docente_nombre: string;
  tipo_accion: string;
  cantidad: number;
}

File: backend/app/schemas/metricas.py
from pydantic import BaseModel, ConfigDict


class MetricasResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")
    total_alumnos: int = 0
    alumnos_activos: int = 0
    porcentaje_riesgo: float = 0.0
    promedio_progreso: float = 0.0
    total_docentes: int = 0
    total_materias_activas: int = 0
    total_carreras_activas: int = 0
